Fix warning C4996 with VS2017

Class Json::Reader is marked as deprecated. Unfortunately, VS2017 issues a bunch of warnings even when client code is not using Json::Reader at all. The warnings refer to usage of Json::Reader *within* Json::Reader itself:
```
...\jsoncpp\reader.h(38): warning C4996: 'Json::Reader::Char': Use CharReader and CharReaderBuilder instead
...\jsoncpp\reader.h(37): note: see declaration of 'Json::Reader::Char'
...\jsoncpp\reader.h(181): warning C4996: 'Json::Reader::Location': Use CharReader and CharReaderBuilder instead
...\jsoncpp\reader.h(38): note: see declaration of 'Json::Reader::Location'
...\jsoncpp\reader.h(182): warning C4996: 'Json::Reader::Location': Use CharReader and CharReaderBuilder instead
[...]
```

This minimal example also results in warnings: 
```
class Foo
{
    typedef char Char; // C4996 WARNING HERE
};

void main()
{
  // not using Foo
}
```

What works for VS2017 is the following:

```
// class declaration without declaring as "deprecated" yet
class Foo
{
    typedef char Char; // C4996 WARNING HERE
};
// now make it deprecated
class __declspec(deprecated("FOOBAR")) Foo;
```

I have *not* tested it whether this also works on other platforms with the JSON_DEPRECATED macro.

--- a/include/json/reader.h
+++ b/include/json/reader.h
@@ -32,7 +32,7 @@ namespace Json {
  *
  * \deprecated Use CharReader and CharReaderBuilder.
  */
-class JSONCPP_DEPRECATED("Use CharReader and CharReaderBuilder instead") JSON_API Reader {
+class JSON_API Reader {
 public:
   typedef char Char;
   typedef const Char* Location;
@@ -52,11 +52,13 @@ public:
   /** \brief Constructs a Reader allowing all features
    * for parsing.
    */
+  JSONCPP_DEPRECATED("Use CharReader and CharReaderBuilder instead") 
   Reader();
 
   /** \brief Constructs a Reader allowing the specified feature set
    * for parsing.
    */
+  JSONCPP_DEPRECATED("Use CharReader and CharReaderBuilder instead") 
   Reader(const Features& features);
 
   /** \brief Read a Value from a <a HREF="http://www.json.org">JSON</a>

